src/Modules/UngradedModules.jl

```diff
@@ -1261,9 +1276,10 @@ function (==)(M::SubModuleOfFreeModule, N::SubModuleOfFreeModule)
     return false
   end
   #TODO should there be a check for === up to permutation in order to avoid std-computation?
-  M_mod_N = reduce(M, N)
-  N_mod_M = reduce(N, M)
-  return iszero(M_mod_N) && iszero(N_mod_M)
+  # If yes, this could also be incorporated in the `in`-function.
+  all(x->(x in N), gens(M)) || return false
+  all(x->(x in M), gens(N)) || return false 
```

Comment:
This does the same but has the advantage that it aborts early?
So this is an attempt at speedup? Can you support this by timing a few examples?

Reply:
Could also just be
```julia
all(x->(x in N), gens(M)) && all(x->(x in M), gens(N))
```

Reply:
No. This was to make the code as generic as it was intended to be. `reduce` tried to do groebner basis stuff which it couldn't.